be/src/olap/schema.h

```diff
@@ -120,39 +119,36 @@ class Schema {
     bool is_null(const char* row, int index) const {
         return *reinterpret_cast<const bool*>(row + _col_offsets[index]);
     }
-
     void set_is_null(void* row, uint32_t cid, bool is_null) const {
         *reinterpret_cast<bool*>((char*)row + _col_offsets[cid]) = is_null;
     }
 
-    size_t schema_size() const {
-        return _schema_size;
-    }
-
     size_t num_columns() const { return _cols.size(); }
     size_t num_column_ids() const { return _col_ids.size(); }
     const std::vector<ColumnId>& column_ids() const { return _col_ids; }
+
 private:
-    // all valid ColumnIds in this schema
+    void _init(const std::vector<TabletColumn>& cols,
+               const std::vector<ColumnId>& col_ids,
+               size_t num_key_columns);
+    void _init(const std::vector<const Field*>& cols,
+               const std::vector<ColumnId>& col_ids,
+               size_t num_key_columns);
+
+    void _copy_from(const Schema& other);
+
+    // NOTE: The ColumnId here represents the sequential index number (starting from 0) of
+    // a column in current row, not the unique id-identifier of each column
```

Comment:
if `_col_ids` is what you mean in comment, why not just using `int32` instead of `ColumnId` ?

Reply:
It seems that the type alias `ColumnId` really means `ColumnIdx` in our code，so It was left there unmodified.